src/k8s/cmd/k8s/k8s_disable.go

```diff
@@ -5,34 +5,77 @@ import (
 	"slices"
 	"strings"
 
+	api "github.com/canonical/k8s/api/v1"
+	"github.com/canonical/k8s/cmd/k8s/errors"
 	"github.com/spf13/cobra"
 )
 
 func newDisableCmd() *cobra.Command {
 	disableCmd := &cobra.Command{
-		Use:   "disable <component>",
-		Short: "Disable a specific component in the cluster",
-		Long:  fmt.Sprintf("Disable one of the specific components: %s.", strings.Join(componentList, ",")),
-		RunE: func(cmd *cobra.Command, args []string) error {
+		Use:               "disable <functionality>",
+		Short:             "Disable a specific functionality in the cluster",
+		Long:              fmt.Sprintf("Disable one of the specific functionalities: %s.", strings.Join(componentList, ",")),
+		PersistentPreRunE: chainPreRunHooks(hookSetupClient),
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			defer errors.Transform(&err, nil)
+
 			if len(args) > 1 {
-				return fmt.Errorf("too many arguments: provide only the name of the component that should be disabled")
+				return fmt.Errorf("too many arguments: provide only the name of the functionality that should be disabled")
 			}
 			if len(args) < 1 {
-				return fmt.Errorf("missing argument: provide the name of the component that should be disabled")
+				return fmt.Errorf("missing argument: provide the name of the functionality that should be disabled")
 			}
 			if !slices.Contains(componentList, args[0]) {
-				return fmt.Errorf("unknown component %q; needs to be one of: %s", args[0], strings.Join(componentList, ", "))
+				return fmt.Errorf("unknown functionality %q; needs to be one of: %s", args[0], strings.Join(componentList, ", "))
+			}
+
+			enabled := false
+
+			config := api.UserFacingClusterConfig{}
+
+			switch args[0] {
+			case "network":
+				config.Network = &api.NetworkConfig{
+					Enabled: &enabled,
```

Comment:
we'll have this a lot, does it make sense to add a util function like

```
func Pointer(v T) *T {
  return &v
}
```

then change all these to `utils.Pointer(false)`?

```suggestion
					Enabled: &enabled,
```

Reply:
why not just `new(bool)`?

Reply:
@bschimke95 I think `new(bool)` would only assign zero value(false)